crates/noirc_evaluator/src/ssa_refactor/ir/map.rs

```diff
@@ -25,6 +25,10 @@ impl<T> Id<T> {
         Self { index, _marker: std::marker::PhantomData }
     }
 
+    pub fn as_usize(&self) -> usize {
+        self.index
+    }
```

Comment:
We should avoid exposing the index of an Id and should treat them as opaque when possible. It looks like the usage of this can be replaced with hashing for example. `HashMap<ValueId, RegisterIndex>` where we'd keep around a counter for the next available register index and store the next value id at that index. Essentially what is already done for `get_tmp_register`